extensions-contrib/compressed-bigdecimal/src/main/java/org/apache/druid/compressedbigdecimal/CompressedBigDecimalAggregator.java

```diff
@@ -54,10 +54,10 @@ public CompressedBigDecimalAggregator(
   @Override
   public void aggregate()
   {
-    CompressedBigDecimal<?> selectedObject = selector.getObject();
+    CompressedBigDecimal selectedObject = Utils.objToCompressedBigDecimal(selector.getObject());
     if (selectedObject != null) {
       if (selectedObject.getScale() != sum.getScale()) {
-        selectedObject = Utils.scaleUp(selectedObject);
+        selectedObject = Utils.scaleUp(selectedObject, sum.getScale());
```

Comment:
Is there a reason not to push the "scale up" part into the `Utils.objToCompressedBigDecimal()` call itself?

Reply:
i don't think so. we should have this function throw an exception if the scales don't match. it doesn't even document that the scales must match. you have to follow into `internalAdd` and see that the scale isn't examined

Reply:
oh and also `Utils.objToCompressedBigDecimal()` is called in the extractor which doesn't presently have the scale. unless  ComplexMetricSerde can have a constructor with params?

Reply:
and also^2, `CBD.accumulate()` already does this, but it seems to use `Utils.accumulate()` to avoid object creation...
(which interestingly makes the generified `cbd.accumulate()` method with various lambdas kind of pointless...it will be monomorphic in practice. I think the method it uses maybe is only used in tests(?)

basically CBD.accumulate() is only ever called on `ArrayCompressedBigDecimal`

Reply:
it might be better to just have 2 impls of `CompressedBigDecimal.accumulate()` and have the buffer version use `Utils.accumulate`....

it would be easier to read code (though i'm a little scaled to change `CompressedBigDecimal.internalAdd` and break it....test coverage *seems* adequate